### PR TITLE
Update dependencies + Theme AppDialog based on the game icon

### PR DESCRIPTION
**Pull request by PabloG02**
Marking it as draft until stable material-components 1.10 is released.

### CODE REVIEW

File: app/src/main/java/emu/skyline/input/onscreen/OnScreenItemDefinitions.kt [hunk 1]
```diff
@@ -94,7 +94,7 @@ open class JoystickButton(
     override fun renderCenteredText(canvas : Canvas, text : String, size : Float, x : Float, y : Float, alpha : Int) = Unit
 
     private val activationRadiusPaint = Paint().apply {
-        color = onScreenControllerView.context.obtainStyledAttributes(intArrayOf(R.attr.colorPrimary)).use { it.getColor(0, Color.RED) }
+        color = onScreenControllerView.context.obtainStyledAttributes(intArrayOf(androidx.appcompat.R.attr.colorPrimary)).use { it.getColor(0, Color.RED) }
```
Comment (lynxnb) [inline app/src/main/java/emu/skyline/input/onscreen/OnScreenItemDefinitions.kt:97]: Incorrect, we don't want to use AndroidX default primary color, but the one from our theme.

File: app/src/main/java/emu/skyline/input/onscreen/OnScreenControllerView.kt [hunk 2]
```diff
@@ -84,7 +84,7 @@ class OnScreenControllerView @JvmOverloads constructor(context : Context, attrs
     }
 
     private val selectionPaint = Paint().apply {
-        color = context.obtainStyledAttributes(intArrayOf(R.attr.colorPrimary)).use { it.getColor(0, Color.RED) }
+        color = context.obtainStyledAttributes(intArrayOf(androidx.appcompat.R.attr.colorPrimary)).use { it.getColor(0, Color.RED) }
```
Comment (lynxnb) [inline app/src/main/java/emu/skyline/input/onscreen/OnScreenControllerView.kt:87]: Incorrect, we don't want to use AndroidX default primary color, but the one from our theme.

File: app/src/main/java/emu/skyline/input/onscreen/OnScreenButton.kt [hunk 3]
```diff
@@ -47,7 +47,7 @@ abstract class OnScreenButton(
     init {
         if (disabledSelectionPaint == null) {
             disabledSelectionPaint = Paint().apply {
-                color = onScreenControllerView.context.obtainStyledAttributes(intArrayOf(R.attr.colorTertiary)).use { it.getColor(0, Color.GREEN) }
+                color = onScreenControllerView.context.obtainStyledAttributes(intArrayOf(com.google.android.material.R.attr.colorTertiary)).use { it.getColor(0, Color.GREEN) }
```
Comment (lynxnb) [inline app/src/main/java/emu/skyline/input/onscreen/OnScreenButton.kt:50]: Incorrect, we don't want to use AndroidX default tertiary color, but the one from our theme.

File: app/src/main/java/emu/skyline/MainActivity.kt [hunk 4]
```diff
@@ -113,8 +113,8 @@ class MainActivity : AppCompatActivity() {
         setupAppList()
 
         binding.swipeRefreshLayout.apply {
-            setProgressBackgroundColorSchemeColor(obtainStyledAttributes(intArrayOf(R.attr.colorSurfaceVariant)).use { it.getColor(0, Color.BLACK) })
-            setColorSchemeColors(obtainStyledAttributes(intArrayOf(R.attr.colorPrimary)).use { it.getColor(0, Color.WHITE) })
+            setProgressBackgroundColorSchemeColor(obtainStyledAttributes(intArrayOf(com.google.android.material.R.attr.colorSurfaceVariant)).use { it.getColor(0, Color.BLACK) })
+            setColorSchemeColors(obtainStyledAttributes(intArrayOf(androidx.appcompat.R.attr.colorPrimary)).use { it.getColor(0, Color.WHITE) })
```
Comment (lynxnb) [inline app/src/main/java/emu/skyline/MainActivity.kt:117]: Incorrect, we don't want to use AndroidX default colors, but the ones from our theme.

**Review by lynxnb — COMMENTED** [review]
This PR is too messy, there's too many commits unrelated to each other, some commits also perform changes unrelated to what the commit title says, so they need to be separated.

Commit [Update target SDK version to Android 14](https://github.com/strato-emu/strato/pull/179/commits/701a0e7ecf1a6516d8f68e188cc039e19ffa3a8a) needs to go on its own PR as bumping the SDK target is not something we want hidden.

Commits [Update AGP to 8.1.1](https://github.com/strato-emu/strato/pull/179/commits/4c638568024de8ef49ad81d052e753e3585b5fea) and [Update Kotlin (1.8.22), Java (17) and Kotlin Dependencies](https://github.com/strato-emu/strato/pull/179/commits/f841ca6cca776018a057434df6a3f669021175a2) should go on their own PR, and the bump to Java 17 should be well separated from the rest, and have its own commit.

Commit [Gradle: Use non-transitive R classes](https://github.com/strato-emu/strato/pull/179/commits/05c44f929ac5d73f50851c1aa04524cabe99bc37) has some errors because of the refactor to non-transitive R classes, see comments below.

Commit [Theme AppDialog based on the game icon](https://github.com/strato-emu/strato/pull/179/commits/fba1acd4246a2e5ce420e6481787332c0bdcc11c) should go on its own PR.

Commit [Fix try-catch in save import](https://github.com/strato-emu/strato/pull/179/commits/76e0f8b5a938e39caf132391a4bc181186103c9b) should go on its own PR.

Commit [Use string resources in ProfilePicturePreference](https://github.com/strato-emu/strato/pull/179/commits/36b2131ecfa3f7d264a5af63ea2397f63f4db038) should go on its own PR.

**Comment by lynxnb** [comment]
It is imperative that changes like these are well separated with each other, otherwise any regression hunting becomes a nightmare.
Commits should be as atomic as possible, and only contain the changes the commit title describes.
PRs should group up commits in a semantic way, where that makes sense. We're not against single-commit PRs.

**Comment by Ishan0981** [comment]
@PabloG02 
I tested the **Theme app dialog based on the game icon** commit and it's made the app dialog more slower so the app dialog is taking more time to open.

**Comment by lynxnb** [comment]
I'm going to close this as it still has a lot of unrelated commits. Feel free to open another PR for the game based AppDialog theme alone.

**Pull request closed without merging**